src/Modules/Admin/Commands.Admin/AddPowerBIEncryptionKey.cs

```diff
@@ -0,0 +1,75 @@
+﻿/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+using System;
+using System.Management.Automation;
+using Microsoft.PowerBI.Common.Client;
+
+namespace Microsoft.PowerBI.Commands.Admin
+{
+    [Cmdlet(CmdletVerb, CmdletName)]
+    public class AddPowerBIEncryptionKey : PowerBIClientCmdlet
+    {
+        public const string CmdletName = "PowerBIEncryptionKey";
+        public const string CmdletVerb = VerbsCommon.Add;
+
+        public AddPowerBIEncryptionKey() : base() { }
+
+        public AddPowerBIEncryptionKey(IPowerBIClientCmdletInitFactory init) : base(init) { }
+
+        #region Parameters
+
+        [Parameter(Mandatory = true)]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true)]
+        public string KeyVaultKeyUri { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public bool Default { get; set; } = false;
+
+        [Parameter(Mandatory = false)]
+        public bool Activate { get; set; } = false;
```

Comment:
Use three Parameter Sets, one for Default and the other Activate and one for both, set the default as the default parameter set. That way they have to declare at least one of the switches and they don't have to pass true\false, they just -SwitchName